chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -780,22 +783,32 @@ func (b *BitcoindNotifier) historicalSpendDetails(
 	spendRequest chainntnfs.SpendRequest, startHeight, endHeight uint32) (
 	*chainntnfs.SpendDetail, error) {
 
-	// Begin scanning blocks at every height to determine if the outpoint
-	// was spent.
+	// batchrequest will store the scan requests at every height to determine
+	// if the output was spent
+	batchRequest := []rpcclient.FutureGetBlockHashResult{}
```

Comment:
This could be a map with the height as the key. And we should pre-allocate as well to avoid it being resized (the number of elements is `endHeight - startHeight`).